astropy/coordinates/sky_coordinate.py

```diff
@@ -501,6 +501,41 @@ def to_string(self, style='decimal', **kwargs):
 
         return coord_string
 
+    def is_equivalent_frame(self, other):
+        """
+        Checks if this object's frame as the same as that of the ``other``
+        object.
+
+        To be the same frame, two objects must be the same frame class and have
+        the same frame attributes. For two `SkyCoord` objects, *all* of the
+        frame attributes have to match, not just those relevant for the object's
+        frame.
+
+        Parameters
+        ----------
+        other : SkyCoord or BaseCoordinateFrame
+            The other object to check.
+
+        Returns
+        -------
+        isequiv : bool
+            True if the frames are the same, False if not.
+        """
+        if isinstance(other, BaseCoordinateFrame):
+            return self.frame.is_equivalent_frame(other)
+        elif hasattr(other, 'frame'):
+            # assume it's a SkyCoord-ish thing
+            if other.frame.name != self.frame.name:
+                return False
+
+            for fattrnm in FRAME_ATTR_NAMES_SET():
+                if getattr(self, fattrnm) != getattr(other, fattrnm):
+                    return False
+            return True
+        else:
+            #not a BaseCoordinateFrame nor a SkyCoord-ish object
+            return False
```

Comment:
Hmm, that's a good point.  I was initially thinking this is like ``1 == 1.1`` where you still want that to be False even though the types are different.  But that's *not* the same as this, because the method actually does say "frame" pretty clearly (and it doesn't really make sense any other way).  So I'll change it to do `TypeError`.